src/web/error.rs

```diff
@@ -132,6 +136,110 @@ impl From<PoolError> for IronError {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+#[allow(dead_code)] // FIXME: remove after iron is gone
+pub enum AxumNope {
+    #[error("Requested resource not found")]
+    ResourceNotFound,
+    #[error("Requested build not found")]
+    BuildNotFound,
+    #[error("Requested crate not found")]
+    CrateNotFound,
+    #[error("Requested owner not found")]
+    OwnerNotFound,
+    #[error("Requested crate does not have specified version")]
+    VersionNotFound,
+    #[error("Search yielded no results")]
+    NoResults,
+    #[error("Internal server error")]
+    InternalServerError,
+    #[error("internal error")]
+    InternalError(#[from] anyhow::Error),
+    #[error("internal threading error")]
+    InternalTokio(#[from] tokio::task::JoinError),
+}
+
+impl IntoResponse for AxumNope {
+    fn into_response(self) -> AxumResponse {
+        match self {
+            AxumNope::ResourceNotFound => {
+                // user tried to navigate to a resource (doc page/file) that doesn't exist
+                AxumErrorPage {
+                    title: "Requested resource does not exist",
+                    message: Some("no such resource".into()),
+                    status: StatusCode::NOT_FOUND,
+                }
+                .into_response()
+            }
+
+            AxumNope::BuildNotFound => AxumErrorPage {
+                title: "The requested build does not exist",
+                message: Some("no such build".into()),
+                status: StatusCode::NOT_FOUND,
+            }
+            .into_response(),
+
+            AxumNope::CrateNotFound => {
+                // user tried to navigate to a crate that doesn't exist
+                // TODO: Display the attempted crate and a link to a search for said crate
+                AxumErrorPage {
+                    title: "The requested crate does not exist",
+                    message: Some("no such crate".into()),
+                    status: StatusCode::NOT_FOUND,
+                }
+                .into_response()
+            }
+
+            AxumNope::OwnerNotFound => AxumErrorPage {
+                title: "The requested owner does not exist",
+                message: Some("no such owner".into()),
+                status: StatusCode::NOT_FOUND,
+            }
+            .into_response(),
+
+            AxumNope::VersionNotFound => {
+                // user tried to navigate to a crate with a version that does not exist
+                // TODO: Display the attempted crate and version
+                AxumErrorPage {
+                    title: "The requested version does not exist",
+                    message: Some("no such version for this crate".into()),
+                    status: StatusCode::NOT_FOUND,
+                }
+                .into_response()
+            }
+            AxumNope::NoResults => {
+                todo!("to be implemented when search-handler is migrated to axum")
```

Comment:
is this hard to fix? I would prefer not to panic anywhere ...

Reply:
This error is rendered via the `Search` page, not the `Error` page, 

so fixing this would mean duplicating the search-page, so I thought it easier to just migrate the error together with the search-page. 

So I just commented out the variant, to be readded when we migrate the search-handler

Reply:
btw, 
> I would prefer not to panic anywhere ... 

I fully agree on this, especially seeing all the `.unwrap` in the current web handlers